service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java

```diff
@@ -0,0 +1,258 @@
+package bio.terra.axonserver.utils.dataproc;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.withSettings;
+
+import bio.terra.axonserver.service.exception.ComponentNotFoundException;
+import bio.terra.cloudres.google.dataproc.ClusterName;
+import bio.terra.cloudres.google.dataproc.DataprocCow;
+import com.google.api.services.dataproc.model.Cluster;
+import com.google.api.services.dataproc.model.ClusterConfig;
+import com.google.api.services.dataproc.model.EndpointConfig;
+import com.google.api.services.dataproc.model.Operation;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.ws.rs.InternalServerErrorException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.MockMakers;
+import org.mockito.MockSettings;
+import org.mockito.Mockito;
+
+public class GoogleDataprocClusterTest {
+  private final ClusterName clusterName =
+      ClusterName.builder()
+          .projectId("fakeproject")
+          .region("fakeregion")
+          .name("fakeclustername")
+          .build();
+
+  /**
+   * Settings required to mock final classes/methods, required for several GCP SDK objects. Requires
```

Comment:
never seen this. is this for dataprocCow.cluster and Operation?

Reply:
Following a pattern John C dith sagemaker+vertex ai nb operations